### PR TITLE
Export single ith test histogram generation functions

### PR DESCRIPTION
**Pull request by krajorama**
Make it easier to reuse test histogram generation.

### CODE REVIEW

File: tsdb/head.go [hunk 1]
```diff
@@ -2040,106 +2040,118 @@ func (h *Head) updateWALReplayStatusRead(current int) {
...
+		PositiveBuckets: []int64{int64(i + 1), 1, -1, 0},
+		NegativeSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
+	}
+	if i > 0 {
+		h.CounterResetHint = histogram.NotCounterReset
+	}
```
Comment (codesome) [inline tsdb/head.go:2068]: Looking from a perspective of exported functions, this should go in `GenerateTestHistograms`. 

`GenerateTestHistogram`, from an external perspective, just gives a test histogram, and the user has to decide the counter reset header.

File: tsdb/head.go [hunk 2]
```diff
@@ -2040,106 +2040,118 @@ func (h *Head) updateWALReplayStatusRead(current int) {
...
-			},
-			NegativeBuckets: []int64{int64(i + 1), 1, -1, 0},
-		})
+		r = append(r, GenerateTestGaugeHistogram(rand.Intn(n)))
 	}
 	return r
 }
 
+func GenerateTestGaugeHistogram(i int) *histogram.Histogram {
+	return &histogram.Histogram{
```
Comment (codesome) [inline tsdb/head.go:2080]: Now that we have `GenerateTestHistogram`, we can use that here and set the CounterResetHint before returning. Both are having the same histogram.

File: tsdb/head.go [hunk 3]
```diff
@@ -2040,106 +2040,118 @@ func (h *Head) updateWALReplayStatusRead(current int) {
...
-			NegativeBuckets: []float64{float64(i + 1), float64(i + 2), float64(i + 1), float64(i + 1)},
-		})
+		r = append(r, GenerateTestGaugeFloatHistogram(rand.Intn(n)))
 	}
-
 	return r
 }
+
+func GenerateTestGaugeFloatHistogram(i int) *histogram.FloatHistogram {
+	return &histogram.FloatHistogram{
```
Comment (codesome) [inline tsdb/head.go:2139]: Similar comment about re-using `GenerateTestFloatHistogram` here

File: tsdb/head.go [hunk 4]
```diff
@@ -2040,106 +2040,118 @@ func (h *Head) updateWALReplayStatusRead(current int) {
...
+		PositiveBuckets: []float64{float64(i + 1), float64(i + 2), float64(i + 1), float64(i + 1)},
+		NegativeSpans: []histogram.Span{
+			{Offset: 0, Length: 2},
+			{Offset: 1, Length: 2},
+		},
+		NegativeBuckets: []float64{float64(i + 1), float64(i + 2), float64(i + 1), float64(i + 1)},
+	}
+	if i > 0 {
+		h.CounterResetHint = histogram.NotCounterReset
+	}
```
Comment (codesome) [inline tsdb/head.go:2127]: Similar comment as `GenerateTestHistogram`. Alternatively, we can add a comment to these functions that they set NotCounterReset if i>0.

File: tsdb/head.go (hunk 2 above)
Comment (krajorama) [reply tsdb/head.go:2080]: Yes, I've thought of that, but wasn't sure , working on it.

File: tsdb/head.go (hunk 1 above)
Comment (krajorama) [reply tsdb/head.go]: makes sense, updated

File: tsdb/head.go (hunk 4 above)
Comment (krajorama) [reply tsdb/head.go]: I've added a comment about _not_ setting the counter reset hint .

**Pull request merged**